<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Document</title>
</head>
<body>
    
</body>
<script>
try {

    setTimeout(() => {
        badRunTimeCode;
        console.log("오류가 발생할 수 있는 코드");
    },1000);
}catch (error) {
    alert("에러 핸들링", error);
}finally{
    alert('무조건 한 번 실행하는 finally')
}

    // * 비동기 코드에서 발생한 예외를 동기 코드에서 핸들링할 수 없다.
// setTimeout(() => {
//   try {
//     //한번 실행되고 오류가 발생되서 alert 창에 catch문 내용이 뜬다. 
//     badRunTimeCode;
//     console.log("오류가 발생할 수 있는 코드");
//   } catch (error) {
//     alert('에러 핸들링')
//   }
// }, 1000);
</script>
</html>